src/detection/Footprint.cc

```diff
@@ -1452,6 +1482,15 @@ PTR(Footprint) shrinkFootprintImpl(
     }
 
     shrunk->normalize();
+
+    // Peaks from the original footprint have not yet been added to the shrunken footprint.
+    // Iterate over peaks from the original footprint and add them IF they are contained
+    // within the shrunken footprint.
+    for (auto peakIter = foot.getPeaks().begin(); peakIter != foot.getPeaks().end(); peakIter++) {
+        if (shrunk->contains(geom::Point2I(peakIter->getIx(), peakIter->getIy()))) {
+            shrunk->addPeak(peakIter->getIx(), peakIter->getIy(), peakIter->getPeakValue());
```

Comment:
This doesn't necessarily transfer the entire peak (since it may have fields besides the position and value).  Instead, I think you want:
```
shrunk->getPeaks().addNew()->assign(*peakIter);
```